test/unit/fact_importer_test.rb

```diff
@@ -55,6 +55,73 @@ def fact_name_class
     end
   end
 
+  describe '#normalize' do
+    test 'filters regular facts' do
+      Setting.stubs(:[]).with(:excluded_facts).returns(['ignore*', '*_bad', 'filter'])
+
+      facts = {
+        'ignored_fact' => 'will_not_show',
+        'fact_bad' => 'will_not_show',
+        'filter' => 'will_not_show',
+        'just_a_fact' => 'hello',
+      }
+
+      importer = FactImporter.new(nil, facts)
+      actual_facts = importer.send(:facts)
+
+      assert_not_nil actual_facts['just_a_fact']
+      assert_equal 1, actual_facts.count
+    end
+
+    test 'filters a::b facts' do
+      Setting.stubs(:[]).with(:excluded_facts).returns(['ignore*', '*_bad', 'filter'])
+
+      facts = {
+        # rhsm style flat facts: a::b::c
+        'ignored_fact::something' => 'will_not_show',
+        'something::ignored_fact' => 'will_not_show',
+        'something::ignored_fact::something_else' => 'will_not_show',
+        'fact_bad::something' => 'will_not_show',
+        'something::fact_bad' => 'will_not_show',
+        'something::fact_bad::something_else' => 'will_not_show',
+        'filter::something' => 'will_not_show',
+        'something::filter' => 'will_not_show',
+        'something::filter::something_else' => 'will_not_show',
+        'something::something_else' => 'hello'
+      }
+
+      importer = FactImporter.new(nil, facts)
+      actual_facts = importer.send(:facts)
+
+      assert_not_nil actual_facts['something::something_else']
+      assert_equal 1, actual_facts.count
```

Comment:
Can you also assert that all passed facts have "hello" in its value? Just to be sure it filtered it correctly :-) In all tests please.